### ISSUE TITLE
C# Add New Class is broken when RootNamespace is empty

### ISSUE THREAD

**Issue by tannergooding** [dotnet/project-system#1989]
**Repro**
1. Create a new C# Class Library (.NET Standard)
2. Edit the project properties and set the Default Namespace to nothing
3. Add New Item -> Class (Visual C# Items)
4. Observe the invalid code generated

```C#
using System;
using System.Collections.Generic;
using System.Text;

namespace 
{
    class Class2
    {
    }
}
```

**Comment by tannergooding** [comment]
FYI. @srivatsn, @jaredpar

**Comment by davkean** [comment]
Funnily enough I filed this ~13 years ago before I joined Microsoft, and it was "fixed" by restricting the user from putting an empty Default Namespace via the UI. ie This bug has always existed since Whidbey.

I'm not sure what the real fix is here - put a fake namespace? Don't generate a namespace for the class itself (which would require us to basically ship different templates just for the empty case)?

**Comment by jmarolf** [comment]
We could write a wizard that removes the namespace text if the `rootnamespace` property is empty.  UI-less template wizards can make arbitrary file modifications and read the dictionary of all template parameters.  What does `dotnet new` do if you set the default namespace to nothing?

**Comment by davkean** [comment]
You could probably hack it with the basic #if support, but then you'd also need to format the document afterword.

**Comment by jmarolf** [comment]
In VS I it will auto-format the file on creation.  Don't know if any of that works in the CLI templating engine

**Comment by SolderedMushroom** [comment]
Why not just fall back to the name of the project? That doesn't seem fake to me, considering the "default default" namespace is the project name.

**Comment by davkean** [comment]
@SolderedMushroom Because the this isn't the case where the default namespace is set to the project name, this is the case where the default namespace is literally empty.

**Comment by SolderedMushroom** [comment]
@davkean Right, my point is just that since an empty namespace is invalid, falling back makes sense, and the project name is the "true" default (what you'd get if the default namespace isn't user-specified at all).
